R/utilities-time-profile.R

```diff
@@ -27,16 +27,33 @@ plotQualificationTimeProfiles <- function(configurationPlan,
     simulation <- loadSimulation(simulationFile)
     simulationResults <- importResultsFromCSV(simulation, simulationResultsFile)
 
+    # Get mol weight for observed data as first molecule path
+    moleculePath <- utils::head(ospsuite::getAllMoleculePathsIn(simulation), 1)
+    molWeightForObservedData <- simulation$molWeightFor(moleculePath)

```

Comment:
Why "first molecule path" and not by the molecule name???
The order of molecules in simulation is random

Reply:
The issue is that the path defined by observed data curve does not necessarily match with the molecule name of the simulation. 
For instance, in the first plot of minimal example, the molecule should be "Midazolam" however, the observed data path is "Midazolam 600mg SD|ObservedData|Peripheral Venous Blood|Plasma|Rifampicin|Conc" which would lead to an empty molecule path because "Rifampicin" is not in the simulation. This create an NA value for molWeight and crashes the workflow because `toUnit` cannot convert the mg/l from the source data into umol/l for the display plot.

Consequently, the strategy I adopted was to use and update the molWeight from the simulated data because it uses an actual quantity path which should be for the same molecule.
But I wanted to prevent the loop to break if there is no simulated data to plot, so I started with the first molecule path I can find in the simulation.

I know this is not optimal, what would you suggest to improve this ?

Reply:
Who created this funny minimal example, lol

I would say:
* In the first place, the qualification runner should export a map "Observed dataset name <=> Molweight", which would be read by the reporting engine then
* The reporting engine should estimate MW like this:
      1. Try to get the MW from this map
      2. If it's not there: try to get the MW from the simulation using the **compound name** from the observed data set
      3. If it's not there: set MW to NaN (don't throw an exception)

Later (e.g. when plotting an observed data) exception will be thrown only if MW is NaN AND it's really required (e.g. in cases where observed data is stored in mass units but must be plotted in molar units).

So for now I would suggest to implement steps 2 and 3.

Makes sense?

@msevestre Any thoughts on this?

Reply:
@Yuri05 
In the case of the exception, how the error should be handled:
1- Throw an actual error and stop the workflow
2- Throw a warning and use the observed data as is (e.g. message can be something like : `Molecular weight required for observed data Id 'Midazolam 600mg SD', but not found. Observed data values were used as is`)
3- Throw a warning and don't use the data (e.g. message can be something like : `Molecular weight required for observed data Id 'Midazolam 600mg SD', but not found. Observed data not used in time profile plot`)

Reply:
My suggestion is:
* If MW cannot be estimated: set it to NaN and show a warning; no exception
* Exception should be thrown only during unit conversion if **MW is required** but is NaN. 
  * E.g. if observed data is given in molar units and must be plotted in molar units as well: MW is not required for the unit conversion, thus even if it's NaN it's not a problem

Reply:
Sorry by exception, I meant the case when MW is NaN and it is required for display unit.

Reply:
>by exception, I meant the case when MW is NaN and it is required for display unit. 

Ok, in this case I would throw an exception and stop the workflow. In such a case there is not enough information provided so the workflow cannot continue. User must either provide a MW then or adjust a qualification plan and exclude this observed data set from the report.